cordapp/workflows/src/main/java/com/newamerica/flows/UpdateFundBalanceFlow.java

```diff
@@ -0,0 +1,137 @@
+import com.newamerica.states.RequestState;
+import net.corda.core.contracts.CommandData;
+import net.corda.core.contracts.ContractState;
+import net.corda.core.contracts.StateAndRef;
+import net.corda.core.contracts.UniqueIdentifier;
+import net.corda.core.crypto.SecureHash;
+import net.corda.core.flows.*;
+import net.corda.core.identity.Party;
+import net.corda.core.node.services.Vault;
+import net.corda.core.node.services.vault.QueryCriteria;
+import net.corda.core.transactions.SignedTransaction;
+import net.corda.core.transactions.TransactionBuilder;
+import net.corda.core.utilities.ProgressTracker;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.newamerica.flows.CordappConfigUtilities.getPreferredNotary;
+import static net.corda.core.contracts.ContractsDSL.requireThat;
+
+public class UpdateFundBalanceFlow {
+
+    @InitiatingFlow
+    @StartableByRPC
+    public static class InitiatorFlow extends FlowLogic<SignedTransaction> {
+        private final StateAndRef requestState;
+        private final UniqueIdentifier fundStateLinearId;
+
+        public InitiatorFlow(
+                StateAndRef requestState,
+                UniqueIdentifier fundStateLinearId
+        ){
+            this.requestState = requestState;
+            this.fundStateLinearId = fundStateLinearId;
+        }
+
+        @Suspendable
+        @Override
+        public SignedTransaction call() throws FlowException {
+            List<UUID> fundStateLinearIdList = new ArrayList<>();
+            fundStateLinearIdList.add(fundStateLinearId.getId());
+
+            //get StatAndRef for the respective FundState
+            QueryCriteria queryCriteria = new QueryCriteria.LinearStateQueryCriteria(null, fundStateLinearIdList);
+            Vault.Page results = getServiceHub().getVaultService().queryBy(FundState.class, queryCriteria);
+            StateAndRef inputStateRef = (StateAndRef) results.getStates().get(0);
+            FundState inputStateRefFundState = (FundState) inputStateRef.getState().getData();
+
+            //create new output state for the fundState
+            RequestState approvedRequestState = (RequestState)inputStateRef.getState().getData();
+            FundState outputFundState = inputStateRefFundState.withdraw(approvedRequestState.getAmount());
+
+            final Party notary = getPreferredNotary(getServiceHub());
+            TransactionBuilder transactionBuilder = new TransactionBuilder(notary);
+            CommandData commandData = new FundContract.Commands.Withdraw();
+            outputFundState.getParticipants().add(getOurIdentity());
+            transactionBuilder.addCommand(commandData, outputFundState.getParticipants().stream().map(i -> (i.getOwningKey())).collect(Collectors.toList()));
+            transactionBuilder.addOutputState(outputFundState, FundContract.ID);
+            transactionBuilder.addReferenceState(requestState.referenced());
```

Comment:
Not sure if we need this since the request state already has a reference to the fund state

Reply:
![image](https://user-images.githubusercontent.com/14203015/91340324-d2c57f00-e7a5-11ea-96e9-51e451073319.png)

Reply:
Removed the referenceState due to privacy concerns.